src/is/L42/flyweight/P.java

```diff
@@ -48,7 +60,7 @@ default boolean wf() {
 
   static P.NCs of(int n, List<C> cs) {
     assert n >= 0;
-    return new NCs(n, cs);
+    return NCs.newNCs(n, cs);

```

Comment:
Should the interface P become sealed? does it give any issue since one kind of P is the enum?